### PR TITLE
Fix bug where Function Emulator debug mode didn't support --inspect-functions on project w/ multiple codebases.

### PR DESCRIPTION
**Pull request by taeold**
In debug mode, enabled by --inspect-functions flag, Function Emulator shares a single worker process across different functions to make it possible to step through function execution from a single debug connection.

Unfortunatley, I didn't foresee that support for multiple codebases will break this - each codebase needs its own runtime process!

Fix here proposes setting up a separate worker pool per codebase.

The PR also includes addition of a new integration test that exercises the scenario of starting the emulator with `--inspect-functions` flag. I made this a separate test target as I found repeatedly starting/shutting down an emulator a flakey process.

### CODE REVIEW

**Comment by taeold** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4903** <a href="https://app.graphite.dev/github/pr/firebase/firebase-tools/4903" target="_blank">Graphite</a> 
    * **PR #4861** <a href="https://app.graphite.dev/github/pr/firebase/firebase-tools/4861" target="_blank">Graphite</a> 
      * **PR #4867** <a href="https://app.graphite.dev/github/pr/firebase/firebase-tools/4867" target="_blank">Graphite</a> 
      * **PR #4949** <a href="https://app.graphite.dev/github/pr/firebase/firebase-tools/4949" target="_blank">Graphite</a>  👈
        * **PR #4886** <a href="https://app.graphite.dev/github/pr/firebase/firebase-tools/4886" target="_blank">Graphite</a> 
          * **PR #4947** <a href="https://app.graphite.dev/github/pr/firebase/firebase-tools/4947" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/firebase/firebase-tools/4949?utm_source=stack-comment).

File: scripts/triggers-end-to-end-tests/tests.inspect.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    test = new TriggerEndToEndTest(FIREBASE_PROJECT, __dirname, config);
+    await test.startEmulators(["--only", "functions", "--inspect-functions"]);
+  });
+
+  after(async function (this) {
+    this.timeout(EMULATORS_SHUTDOWN_DELAY_MS);
+    await test.stopEmulators();
+  });
+
+  it("should invoke correct function in the same codebase", async function (this) {
```
Comment (TheIronDev) [inline scripts/triggers-end-to-end-tests/tests.inspect.ts:39]: Nit (feel free to ack):

is there a reason for using `function` instead of phat arrows?

**Review by TheIronDev — APPROVED** [review]
LGTM!

File: scripts/triggers-end-to-end-tests/tests.inspect.ts (hunk 1 above)
Comment (taeold) [reply scripts/triggers-end-to-end-tests/tests.inspect.ts:39]: yes! Turns out, `this` isn't available if you use arrow fns.

**Pull request merged**